python/tank/commands/setup_project_params.py

```diff
@@ -744,14 +852,39 @@ def pre_setup_validation(self):
         if self._config_template is None:
             raise TankError("Configuration template has not been specified!")
 
-        if self._config_path is None:
+        if self._distribution_mode == self.CENTRALIZED_CONFIG and self._config_path is None:
             raise TankError("Path to the target configuration install has not been specified!")
 
         if self._project_disk_name is None:
             raise TankError("Project disk name has not been specified!")
 
-        # get the location of the configuration
-        config_path_current_os = self.get_configuration_location(sys.platform)
+        # ensure an uploaded_config field exists if we are doing a distributed cfg
+        if self._distribution_mode == self.DISTRIBUTED_CONFIG:
+            # if the fields don't exist in shotgun, the fields will be not be returned
+            field_data = self._sg.schema_field_read("PipelineConfiguration")
+            if "uploaded_config" not in field_data:
+                raise TankError(
+                    "Shotgun site is missing a PipelineConfiguration.uploaded_config "
```

Comment:
centralized distribution mode will raise an exception if executed on older shotguns.

Reply:
Should we mention that they can contact support to get the feature turned on if on a recent enough site?

Reply:
I think that's overkill.